chore: Bump tsconfig target from es2016 to es2020

## Problem

The root `tsconfig.json` has `target: "es2016"`, but the actual browser compatibility constraint is set by Vite's `build.target`, which defaults to `"modules"` (≈ es2020: Chrome 87, Firefox 78, Safari 14, Edge 88). The TypeScript target is more conservative than the real output target with no practical benefit.

The mismatch causes TS to reject valid modern syntax (e.g. BigInt literals `1000n`) even though esbuild would handle them correctly for the targeted browsers.

## Evidence

Encountered while working on PR #3943: using BigInt arithmetic for nanosecond timestamp conversion produced:

```
error TS2737: BigInt literals are not available when targeting lower than ES2020.
```

Even though Vite/esbuild targets es2020-era browsers and would have compiled the output correctly.

## Fix

Bump `tsconfig.json`:

```diff
-    "target": "es2016",
+    "target": "es2020",
```

## Impact

- **Browser compatibility**: None. esbuild (via Vite) controls the final output target, not tsc. The tsconfig `target` only affects what syntax tsc will emit/allow before esbuild processes it.
- **`lib`**: Already set to `"es2022"`, so es2020 APIs are already type-checked. This change only aligns the syntax target with the API target.
- **Node/test environments**: Tests run through Vitest which uses esbuild internally — not affected.
- **Unlocks**: BigInt literals, `String.matchAll`, `Promise.allSettled`, `globalThis` (already usable via `lib`, now also valid syntax for tsc).

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,7 +1,7 @@
 {
   "compileOnSave": true,
   "compilerOptions": {
-    "target": "es2016",
+    "target": "es2020",
     "lib": ["es2022", "dom", "dom.iterable", "webworker"],
     "allowJs": true,
     "skipLibCheck": true,

